lib/validate.js

```diff
@@ -94,8 +94,8 @@ exports.usingChecksum = (deviceFileDescriptor, options = {}) => {
     imageSize: options.imageSize,
     progress: options.progress,
     chunkSize: options.chunkSize
-  }).then((deviceChecksum) => {
-    if (options.imageChecksum !== deviceChecksum) {
+  }).then((results) => {
+    if (options.imageChecksum !== results.crc32) {
```

Comment:
> I guess to remain backwards compatible we could inspect the argument type? 

I don't think we should worry about this. If a change makes sense as a breaking change, we should definitely go for it. This module is an "internal" module mainly targeted at Etcher, so I think we can break compatibility when needed.